File: airesources/Rust/MyBot.rs
mod networking;

use std::collections::HashMap;
use rand::Rng;

fn main() {
	let (myID, mut gameMap) = networking::get_init();
	let mut rng = rand::thread_rng();
	networking::send_init("RustBot".to_string() + myID.to_string());
	loop {
		let mut moves = HashMap::new();
		for y in 0..gameMap.height {
			for x in 0..gameMap.width {
				let l = Location { x, y };
				if gameMap.get_site(l, hlt::STILL).owner == myID {
					moves.insert(l, (rng.gen::<u32>() % 5) as u8);
				}
			}
		}
	}
}

File: airesources/CSharp/MyBot.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class MyBot
{
    public const string MyBotName = "MyC#Bot";

    public static void Main(string[] args) {
        Console.SetIn(Console.In);
        Console.SetOut(Console.Out);

        ushort myID;
        var map = Networking.getInit(out myID);

        /* ------
            Do more prep work, see rules for time limit
        ------ */

        Networking.SendInit(MyBotName); // Acknoweldge the init and begin the game

        var random = new Random();
        while (true) {
            Networking.getFrame(ref map); // Update the map to reflect the moves before this turn

            var moves = new List<Move>();
            for (ushort x = 0; x < map.Width; x++) {
                for (ushort y = 0; y < map.Height; y++) {
                    if (map[x, y].Owner == myID) {
                        moves.Add(new Move {
                            Location = new Location {X = x, Y = y},
                            Direction = (Direction)random.Next(5)
                        });
                    }
                }
            }

            Networking.SendMoves(moves); // Send moves
        }
    }
}
